src/Context/authContext.js

```diff
@@ -1,90 +1,127 @@
-                localStorage.setItem("tokenNotesApp", response.data.encodedToken);
-                authDispatch({type: "LOGIN_SUCCESS", payload: {toastMessage: "Signed up", name: response.data.createdUser.firstName, email: response.data.createdUser.email, userID:response.data.createdUser._id }})
-                  
-        }catch(error){
-            console.log(error);
-            authDispatch({type:"HANDLER_ERROR", payload:{toastMessage: "Handler error"}})
-        }
-
+  toastData: {
+    display: false,
+    data: "",
+    status: "",
+  },
+  signedIn: false,
+  userName: "",
+  userEmail: "",
+  userID: "",
+  notes: [],
+  archivedNotes: [],
+  trashedNotes: [],
+};
+
+const AuthContext = createContext(initialAuthData);
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, initialAuthData);
+
+  const signUp = async (userDetails) => {
+    console.log("userDetaisl:", userDetails);
+    try {
+      const response = await axios.post("/api/auth/signup", {
+        firstName: userDetails.firstName,
+        lastName: userDetails.lastName,
+        email: userDetails.email,
+        password: userDetails.password,
+      });
+      console.log("response:", response.data);
+      localStorage.setItem("tokenNotesApp", response.data.encodedToken);
+      authDispatch({
+        type: "LOGIN_SUCCESS",
+        payload: {
+          toastMessage: "Signed up",
+          name: response.data.createdUser.firstName,
+          email: response.data.createdUser.email,
+          userID: response.data.createdUser._id,
+        },
+      });
+    } catch (error) {
+      console.log(error);
+      authDispatch({
+        type: "HANDLER_ERROR",
+        payload: { toastMessage: "Handler error" },
+      });
     }
-
-  
-    const login = async(userDetails) => {
-        try{
-            const response = await axios.post("/api/auth/login", {email: userDetails.email, password: userDetails.password})
-            
-            localStorage.setItem("tokenNotesApp", response.data.encodedToken)
-            authDispatch({type:"LOGIN_SUCCESS", payload:{toastMessage: "Logged In", name: response.data.foundUser.firstName, email: response.data.foundUser.email, userID: response.data.foundUser._id}})
-            
-        }catch(error){
-            console.log(error)
-            authDispatch({type:"HANDLER_ERROR", payload:{toastMessage: "Handler error"}})
-        }
+  };
+
+  const login = async (userDetails) => {
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email: userDetails.email,
+        password: userDetails.password,
+      });
+
+      localStorage.setItem("tokenNotesApp", response.data.encodedToken);
+      authDispatch({
+        type: "LOGIN_SUCCESS",
+        payload: {
+          toastMessage: "Logged In",
+          name: response.data.foundUser.firstName,
+          email: response.data.foundUser.email,
+          userID: response.data.foundUser._id,
```

Comment:
instead of writing `response.data.foundUser` multiple times u can extract `firstName`  `email `  `_id` 
`const {firstName,email,_id} = response.data.foundUser` Like this way

Reply:
> instead of writing `response.data.foundUser` multiple times u can extract `firstName` `email ` `_id` `const {firstName,email,_id} = response.data.foundUser` Like this way

Will destructure it.